src/NHibernate/Linq/Visitors/ExpressionParameterVisitor.cs

```diff
@@ -161,6 +180,15 @@ protected override Expression VisitConstant(ConstantExpression expression)
 			return base.VisitConstant(expression);
 		}
 
+		private NamedParameter CreateParameter(ConstantExpression expression, object value, IType type)
+		{
+			return new NamedParameter(
+				"p" + (_parameters.Count + 1),
+				value,
+				type,
+				_collectionParameters.Contains(expression));
```

Comment:
I don't really understand. Why we cannot simply use the same check as it was in ExpressionKeyVisitor.VisitParameter?

Reply:
In order to support types that implements `IEnumerable`, which should not be treated as collections. One example is the NodaTime integration that @gliljas is working on (see [above comment](https://github.com/nhibernate/nhibernate-core/pull/2365#issuecomment-626370794)).